code/modules/power/generators/turbine.dm

```diff
@@ -388,12 +447,19 @@
 
 	switch(action)
 		if("toggle_power")
-			if(compressor?.turbine)
+			var/time_until_done = compressor.time_until_overheat_done()
+			if(time_until_done)
+				compressor.starter = FALSE
+				to_chat(usr, "<span class='alert'>The turbine is overheating, please wait [time_until_done / 10] seconds for cooldown procedures to complete.</span>")
+				playsound(src, 'sound/effects/electheart.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
```

Comment:
Minor thing but we should still have a  `. = TRUE` for this `if` block, so that the UI updates instantly.

Reply:
Fixed in latest!